ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js

```diff
@@ -512,24 +500,35 @@ export default class ConfirmApproveContent extends Component {
             blockExplorerUrl: rpcPrefs?.blockExplorerUrl ?? null,
           },
         );
-        const unknownNFTLink = (
+        const blockExplorerElement = (
           <>
             <a
-              href={unknownNFTBlockExplorerLink}
+              href={blockExplorerLink}
               target="_blank"
               rel="noopener noreferrer"
-              className="confirm-approve-content__unknown-asset"
+              title={tokenAddress}
+              className="confirm-approve-content__approval-asset-link"
             >
-              {t('nft')}
+              {titleTokenDescription}
             </a>
             {tokenIdWrapped && <span>{tokenIdWrapped}</span>}
           </>
         );
-        titleTokenDescription = unknownNFTLink;
+        return blockExplorerElement;
       }
     }
 
-    return titleTokenDescription;
+    return (
+      <span
+        className="confirm-approve-content__approval-asset-title"
+        onClick={() => {
```

Comment:
This wouldn't be accessible from a keyboard perspective; could this be a `button`?